From 2b6cc1b90fe73d178b516d04fcb0b6a2c6af77f4 Mon Sep 17 00:00:00 2001
From: Samsuik <40902469+Samsuik@users.noreply.github.com>
Date: Fri, 7 Jul 2023 18:36:10 +0100
Subject: [PATCH] Use THashSet in HashTreeSet

See: https://github.com/PaperMC/Paper/pull/923

NOTE: The reason I am using a trove set instead of fastutil here is that
fastutil has an auto compaction and trove allows easily disabling it.

diff --git a/src/main/java/org/bukkit/craftbukkit/util/HashTreeSet.java b/src/main/java/org/bukkit/craftbukkit/util/HashTreeSet.java
index 80a5c29f3..99ee8bc42 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/HashTreeSet.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/HashTreeSet.java
@@ -8,11 +8,11 @@ import java.util.TreeSet;
 
 public class HashTreeSet<V> implements Set<V> {
 
-    private HashSet<V> hash = new HashSet<V>();
+    private gnu.trove.set.hash.THashSet<V> hash = new gnu.trove.set.hash.THashSet<>(); // Blossom
     private TreeSet<V> tree = new TreeSet<V>();
 
     public HashTreeSet() {
-
+        hash.setAutoCompactionFactor(0.0f); // Blossom
     }
 
     @Override
-- 
2.40.0.windows.1

